support-frontend/app/controllers/Application.scala

```diff
@@ -102,20 +94,28 @@ class Application(
   }
 
   def contributionsLanding(
-    countryCode: String
-  ): Action[AnyContent] = maybeAuthenticatedAction().async { implicit request =>
+                            countryCode: String,
+                            campaignCode: String
+                          ): Action[AnyContent] = maybeAuthenticatedAction().async { implicit request =>
+
     type Attempt[A] = EitherT[Future, String, A]
-    implicit val settings: AllSettings = settingsProvider.getAllSettings()
-    request.user.traverse[Attempt, IdUser](identityService.getUser(_)).fold(
-      _ => Ok(contributionsHtml(countryCode, None)),
-      user => Ok(contributionsHtml(countryCode, user))
-    ).map(_.withSettingsSurrogateKey)
+
+    val campaignCodeExistsButIsInvalid = campaignCode != "" && campaignCode != "toxicamerica"
```

Comment:
what happens if in future we add a new campaign in the JS but forget to update this line?
Is it necessary to do this server-side check?

Reply:
@tomrf1 we refactored to move the 'arbitrary' bit to the routes file, so it should be more obvious how it's working